include/raft.h.in

```diff
@@ -1022,6 +1022,153 @@ RAFT_API void raft_seed(struct raft *r, unsigned random);
  * The @update output parameter contains changes that user code should perform
  * after the raft_step() call returns (e.g. new entries that must be persisted,
  * new messages that must be sent, etc.).
+ *
+ * Memory Ownership Model for raft_event Fields
+ * =============================================
+ *
+ * The caller must understand the memory ownership semantics for each event type
+ * to avoid memory leaks or use-after-free bugs. This section documents who owns
+ * the memory for each field in the raft_event union after raft_step() returns.
+ *
+ * RAFT_START Event
+ * ----------------
+ * - event->start.term: Scalar value (no ownership concerns)
+ * - event->start.voted_for: Scalar value (no ownership concerns)
+ * - event->start.metadata: PARTIAL OWNERSHIP TRANSFER. The nested configuration
+ *   field is transferred to raft (moved into r->configuration). The metadata
+ *   structure itself is NOT freed by raft and must be freed by the caller after
+ *   raft_step() returns, regardless of success or error. The caller must NOT
+ *   close the configuration field after the call, as it has been transferred.
```

Comment:
This is correct. However, I'm wondering if for consistency and clarity we should rather make a copy of the nested configuration, and leave the ownership of the metadata structure entirely to the caller, so there's no "partial ownership" or ambiguity.